### PR TITLE
Issue 145

### PR DESCRIPTION
**Pull request by TimColaneri**
issue-145 fix

Added a clause to the walker for handling pattern-typematch subsumption cases. The redundant pattern detector will now appropriately handle these cases.

A new regression test, test134.ast, was added to demonstrate that this is functioning correctly.

### CODE REVIEW

File: asteroid/globals.py [hunk 1]
```diff
@@ -167,3 +167,20 @@ def __str__(self):
     'struct',
     'typematch',
 ]
+
+###########################################################################################
+# list of structures that a pattern type-pattern will subsume/overlap
+
+pattern_subsumes = [
```
Comment (lutzhamel) [inline asteroid/globals.py:174]: Your `pattern_subsumes` and the `patterns` list only differ by one element.

I would make `patterns` a set and then define `pattern_subsumes` as,
```
pattern_subsumes = patterns - {'id'}
```
that way we avoid repeated definitions of almost the same things in the code.

Comment (TimColaneri) [reply asteroid/globals.py]: This should be addressed and updated now.

**Review by lutzhamel — APPROVED** [review]
looks good!

**Comment by lutzhamel** [comment]
@TimColaneri feel free to merge this PR and close the associated issue.

**Pull request merged**